include/ensmallen_bits/demon_adam/demon_adam.hpp

```diff
@@ -0,0 +1,206 @@
+/**
+ * @file demon_adam.hpp
+ * @author Marcus Edel
+ *
+ * Definition of DemonAdam.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_DEMON_ADAM_DEMON_ADAM_HPP
+#define ENSMALLEN_DEMON_ADAM_DEMON_ADAM_HPP
+
+#include <ensmallen_bits/sgd/sgd.hpp>
+#include "demon_adam_update.hpp"
```

Comment:
I'll update the includes today.

Reply:
We may need to sort out which format to use (`<xyz>` vs `"xyz"`). 

Use of `<...>` caused problems within the context of RcppArmadillo (see for example https://github.com/RcppCore/RcppArmadillo/pull/281, https://github.com/RcppCore/RcppArmadillo/pull/315, and related issues). The tldr is that for people that had two versions of Armadillo (one via RcppArmadillo, and one via plain Armadillo), there were clashes between the headers from the two versions.  We settled on `"xyz"` to avoid this.

Within the context of ensmallen, rather than doing something like `#include <ensmallen_bits/sgd/sgd.hpp>`, perhaps we can use `..` to specify one level back in the directory structure, ie. `#include "../sgd/sgd.hpp"`.  Alternatively, rather than having subdirectories, we have a flat layout in `ensmallen_bits`, ie. avoid use of further subdirectories.  While the latter option may seem more messy, it has the advantage of all files being visible at the same time.

Reply:
My only opinion on the format is that it's consistent. :) It sounds like `""` is the right way to go here. How should we document this to ensure that it doesn't happen again? Style guide? Build check that greps for `#include <ensmallen_bits/`?

Reply:
That also mean, we I should revert https://github.com/mlpack/ensmallen/pull/234/files/7776391f212b6c20a80c5a7fdb58785625736451#r810741863 and use `"xyz"` instead and adapt `<ensmallen_bits/sgd/sgd.hpp>` accordingly.

I'd like to keep the sub-directory structure, I think it's cleaner, and I don't mind to go one level back and forth. On the GitLab Armadillo repository page, the browser search function is unusable for me; the interface only lists the files in the viewport, so if I search for a file, I have to scroll down to the end, then I can search for a specific file.